packages/core/src/actions/execute.ts

```diff
@@ -721,23 +973,27 @@ export const runEntireDeploymentProcess = async (
       approvalLeafWithProof,
       executionMode,
       provider,
-      signer
+      ownerSignatures,
+      deploymentContext
     )
     ethersReceipts.push(approvalReceipt)
 
-    spinner.succeed(`Approved deployment.`)
+    spinner?.succeed(`Approved deployment.`)
   } else if (merkleRootState.status !== MerkleRootStatus.APPROVED) {
-    spinner.clear()
-    throw new Error(
-      `Deployment's status: ${stringifyMerkleRootStatus(
-        merkleRootState.status
-      )}`
+    return handleStatus(
+      merkleRootState.status,
+      [],
+      ethersReceipts,
+      humanReadableActions[chainId.toString()][
+        Number(merkleRootState.leavesExecuted) - 2
+      ],
+      spinner
```

Comment:
Well we shouldn't be throwing errors within the `compilerAndExecuteDeployment` function. We need to call `deploymentContext.throwError` so that the error can be handled correctly depending on the context. Additionally, the status not being `APPROVED` in this situation doesn't mean we should throw an error at all. If we're repeatedly running the `compileAndExecuteDeployment` function (such as is the case when retrying), then the status could very well be either `COMPETED` or `FAILED` both of which require different handling. 

In the case of `FAILED`, we handle that after this function returns which is why we're just returning `handleStatus` here. In the case of `COMPLETED`, we want to just continue so we can do verification and artifact generation. The only case we don't handle is if the status is `CANCELED` which should not be the case unless the user manually intervenes while the deployment is happening. We should probably add handling for that, but it's outside the scope of this PR IMO.